lib/rules/nonblock-statement-body-position.js

```diff
@@ -104,6 +105,15 @@ module.exports = {
                     validateStatement(node.alternate, "else");
                 }
             },
+            ArrowFunctionExpression(node) {
+
+                // Check if the arrow function body is parenthesized
+                if (node.body && node.body.extra && node.body.extra.parenthesized) {
+                    return;
+                }
+
+                validateStatement(node.body, "arrowFunction");
```

Comment:
I think this could be nonbreaking if it conditionally called `validateStatement` only when the option was actually present.